python/ray/train/v2/_internal/execution/context.py

```diff
@@ -221,6 +222,17 @@ def report(
         related information and not the worker related actions. This refactor
         would also require the `TrainContextCallback` to be updated as well.
         """
+        if "torch" in sys.modules:
+            from ray.air._internal.torch_utils import contains_tensor
+
+            if contains_tensor(metrics):
+                raise ValueError(
+                    "Passing objects containg Torch tensors as metrics "
+                    "is not supported as it will throw an exception on "
+                    "deserialization. You can either convert the tensors "
+                    "to Python objects or report a `train.Checkpoint` "
+                    "with `ray.train.report` to store your Torch objects."
```

Comment:
"You can either convert the tensors to Python objects (ex: `.numpy()`, `.item()`, etc.) or save tensors as part of the checkpoint files instead."